common/src/main/scala/scalan/util/CollectionUtil.scala

```diff
@@ -222,8 +221,8 @@ object CollectionUtil {
       res
     }
 
-    def filterCast[B:ClassTag](implicit cbf: CanBuildFrom[Source[A], B, Source[B]]): Source[B] = {
-      val b = cbf()
+    def filterCast[B:ClassTag](implicit cbf: Factory[B, Source[B]]): Source[B] = {
+      val b = cbf.newBuilder
```

Comment:
scala.collection.compat.BuildFrom should be used. I added the missing dependency in #807 and migrated this file. Please rollback changes and use as example in other places.